llvm/include/llvm/Support/TrailingObjects.h

```diff
@@ -301,6 +302,38 @@ class TrailingObjects : private trailing_objects_internal::TrailingObjectsImpl<
         static_cast<BaseTy *>(this), TrailingObjectsBase::OverloadToken<T>());
   }
 
+  // getTrailingObjects() specialization for a single trailing type.
+  using FirstTrailingType =
+      typename std::tuple_element_t<0, std::tuple<TrailingTys...>>;
+
+  const FirstTrailingType *getTrailingObjects() const {
+    static_assert(sizeof...(TrailingTys) == 1,
```

Comment:
why not `enable_if`?

Reply:
I did try that but did not work. enable_if() has to be based on this functions template type and cannot be based on the the TrailingObject classes template types from what I understood. See this example which I refereed to understand the compiler error I got ('enable_if' cannot be used to disable this declaration): https://stackoverflow.com/questions/52077051/sfinae-enable-if-cannot-be-used-to-disable-this-declaration